cmd/worker/shared/main.go

```diff
@@ -37,6 +63,141 @@ func Main() {
 	})
 	allRoutines = append(allRoutines, server)
 
+	// We're all set up now
+	// Respond positively to ready checks
 	close(ready)
+
 	goroutine.MonitorBackgroundRoutines(context.Background(), allRoutines...)
 }
+
+// loadConfigs calls Load on the configs of each of the tasks registered in this binary.
+// All configs will be loaded regardless if they would later be validated - this is the
+// best place we have to manipulate the environment before the call to env.Lock.
+func loadConfigs(tasks map[string]Task) {
+	// Load the worker config
+	config.names = taskNames(tasks)
+	config.Load()
+
+	// Load all other registered configs
+	for _, task := range tasks {
+		for _, config := range task.Config() {
+			config.Load()
+		}
+	}
+}
+
+// validateConfigs calls Validate on the configs of each of the tasks that will be run
+// by this instance of the worker. If any config has a validation error, a fatal log
+// message will be emitted.
+func validateConfigs(tasks map[string]Task) {
+	validationErrors := map[string][]error{}
+	if err := config.Validate(); err != nil {
+		log.Fatalf("Failed to load configuration: %s", err)
+	}
+	if len(validationErrors) == 0 {
```

Comment:
```suggestion
	
	if len(validationErrors) == 0 {
```

I think https://docs.sourcegraph.com/dev/background-information/languages/go#use-new-lines-intentionally applies here